scripts/create-module.js

```diff
@@ -0,0 +1,50 @@
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+// Function to execute shell commands
+const runCommand = (command) => {
+  try {
+    execSync(command, { stdio: 'inherit' });
+  } catch (error) {
+    console.error(`Error executing command: ${command}`, error);
+  }
+};
+
+// Function to create directories
+const createDir = (dirPath) => {
+  if (!fs.existsSync(dirPath)) {
+    fs.mkdirSync(dirPath, { recursive: true });
+    console.log(`Created directory: ${dirPath}`);
+  } else {
+    console.log(`Directory already exists: ${dirPath}`);
+  }
+};
+
+// Main function to generate module and folders
+const createModule = (moduleName) => {
+  const modulePath = `modules/${moduleName}`;
+
+  // Generate module, controller, and service using Nest CLI
+  runCommand(`nest g module ${modulePath}`);
+  runCommand(`nest g controller ${modulePath}`);
+  runCommand(`nest g service ${modulePath}`);
+
+  // Create additional folders inside the module
+  const basePath = path.join(__dirname, '../src', modulePath);
+  createDir(`${basePath}/dto`);
+  createDir(`${basePath}/interfaces`);
```

Comment:
we are using `types.d.ts` file for interfaces, so let's create a types.d.ts file here instead of interfaces folder